src/DebuggerForm.cpp

```diff
@@ -849,6 +849,21 @@ void DebuggerForm::initConnection()
 		"  return $result\n"
 		"}\n"));
 
+	// define 'debug_check_debuggables' proc for internal use
+	comm.sendCommand(new SimpleCommand(
+		"proc debug_check_debuggables { debuggables } {\n"
+		"  set index 0\n"
+		"  set found_debuggables [debug list]\n"
+		"  set result \"\"\n"
+		"  foreach item $debuggables {\n"
+		"    set index [expr $index + 1]\n"
+		"    if { [lsearch $found_debuggables $item] >= 0} {\n"
+		"      append result \"{$index}\"\n"
+		"    }\n"
+		"  }\n"
+		"  return $result\n"
+		"}\n"));
```

Comment:
Details about this proc itself:
* For weird Tcl reasons it's better to always use `{}` in a `expr`. So replace `expr $index + 1` with `expr {$index + 1}`.
* And `set index [expr {$index + 1}]` can be simplified to `incr index`.

So basically for every item in the `debuggables` list you want to know if it exists as a openMSX debuggable. Consider this alternative implementation:
```
proc debug_check_debuggables {debuggables} {
  set all_debuggables [debug list]
  lmap x $debuggables {expr {[lsearch $all_debuggables $x] >= 0}}
}
```
This takes the same input and then replaces every item in this list (using `lmap`) with a boolean (0 or 1).
So afterwards when using this result you don't search for a specific value, instead you check if a specific position in the result list contains 0 or 1.